config.go

```diff
@@ -61,7 +61,10 @@ func newConfig(logger *zap.Logger) (Config, error) {
 	if configFilepath == "" {
 		logger.Info("the env variable '" + envKey + "' is not set, therefore no YAML config will be loaded")
 	} else {
-		err := k.Load(file.Provider(configFilepath), yaml.Parser())
+		err := k.Load(file.Provider(configFilepath), yaml.Parser(), koanf.WithMergeFunc(func(src, dest map[string]interface{}) error {
+			dest = src
+			return nil
+		}))
```

Comment:
This is only called if you actually provide a YAML config file and therefore this does not fix the referenced issue

Reply:
You are right. Thank you for the hint. I missed the fact that no config file was referenced in the linked issue. I just had a similar problem where I wanted to overwrite some values from the config file with individual environment variables.  Should I create a new issue for this?

Reply:
You can add that to the issue, but the fix I proposed there would fix this as well